test/relayserver/RegistrationManager.test.ts

```diff
@@ -0,0 +1,724 @@
+        const workerBalanceAfter = await relayServer.getWorkerBalance(
+          workerIndex
+        );
+
+        expect(managerBalanceAfter).to.be.equal(constants.Zero);
+        expect(workerBalanceAfter).to.be.equal(constants.Zero);
+      });
+
+      it("should emit 'unstaked' when witihdrawing funds", async function () {
+        const { relayManagerAddress } = relayServer.getChainInfo();
+
+        await relayHub.connect(relayOwner).unlockStake(relayManagerAddress);
+        await evmMineMany(unstakeDelay);
+        await relayHub.connect(relayOwner).withdrawStake(relayManagerAddress);
+
+        let unstakedEmitted = false;
+        relayServer.on('unstaked', () => {
+          unstakedEmitted = true;
+        });
+
+        const latestBlock = await provider.getBlock('latest');
+        await relayServer._worker(latestBlock.number);
+
+        expect(unstakedEmitted, 'unstaked not emitted').to.be.true;
+      });
+    });
+
+    describe('HubUnauthorized event', function () {
+      it('should not send balance immediately after unauthorize (before unstake delay)', async function () {
+        const workerBalanceBefore = await relayServer.getWorkerBalance(
+          workerIndex
+        );
+
+        const { relayManagerAddress } = relayServer.getChainInfo();
+
+        await relayHub.connect(relayOwner).unlockStake(relayManagerAddress);
+
+        await evmMineMany(unstakeDelay - 3);
+        const latestBlock = await provider.getBlock('latest');
+
+        const receipt = await relayServer._worker(latestBlock.number);
+        const receipt2 = await relayServer._worker(latestBlock.number + 1);
+
+        expect(receipt.length).to.be.equal(0);
+        expect(receipt2.length).to.be.equal(0);
+
+        const workerBalanceAfter = await relayServer.getWorkerBalance(
+          workerIndex
+        );
+
+        expect(workerBalanceBefore).to.be.equal(workerBalanceAfter);
+      });
+
+      it('send only workers balances to owner (not manager hub) - after unstake delay', async function () {
+        const { relayManagerAddress } = relayServer.getChainInfo();
+
+        await relayHub.connect(relayOwner).unlockStake(relayManagerAddress);
+
+        const managerBalanceBefore = await relayServer.getManagerBalance();
+        const workerBalanceBefore = await relayServer.getWorkerBalance(
+          workerIndex
+        );
+
+        expect(managerBalanceBefore.gt(constants.Zero)).to.be.true;
+        expect(workerBalanceBefore.gt(constants.Zero)).to.be.true;
+
+        const ownerBalanceBefore = await provider.getBalance(
+          relayOwner.address
+        );
+
+        await evmMineMany(unstakeDelay);
+
+        const latestBlock = await provider.getBlock('latest');
+
+        const receipts = await relayServer._worker(latestBlock.number);
+
+        const gasPrice = await provider.getGasPrice();
+
+        // TODO: these two hard-coded indexes are dependent on the order of operations in 'withdrawAllFunds'
```

Comment:
Which indexes are we referring to?

Reply:
I removed the comment, before this had it, since they were sending the index[0] but its not longer necessary since each test its encapsulated.

Reply:
updated